contracts/src/Gateway.sol

```diff
@@ -462,19 +463,24 @@ contract Gateway is IGateway, IInitializable {
     }
 
     // Convert foreign currency to native currency (ROC/KSM/DOT -> ETH)
-    function _convertToNative(UD60x18 exchangeRate, uint256 amount) internal view returns (uint256) {
-        UD60x18 amountFP = convert(amount);
+    function _convertToNative(UD60x18 exchangeRate, UD60x18 amount) internal view returns (uint256) {
         UD60x18 ethDecimals = convert(1e18);
         UD60x18 foreignDecimals = convert(10).pow(convert(uint256(FOREIGN_TOKEN_DECIMALS)));
-        UD60x18 nativeAmountFP = amountFP.mul(exchangeRate).div(foreignDecimals).mul(ethDecimals);
-        uint256 nativeAmount = convert(nativeAmountFP);
-        return nativeAmount;
+        UD60x18 nativeAmount = amount.mul(exchangeRate).div(foreignDecimals).mul(ethDecimals);
+        return convert(nativeAmount);
     }
 
     // Calculate the fee for accepting an outbound message
     function _calculateFee(Costs memory costs) internal view returns (uint256) {
         PricingStorage.Layout storage pricing = PricingStorage.layout();
-        return costs.native + _convertToNative(pricing.exchangeRate, pricing.deliveryCost + costs.foreign);
+
+        // The exchange rate and amount are both multiplied by `multiplier` to add a safety margin
+        // * `exchangeRate` is subject to real-world fluctuations
+        // * `amount` includes XCM execution fees in Polkadot that are also subject to fluctuations
+        UD60x18 exchangeRate = pricing.multiplier.mul(pricing.exchangeRate);
+        UD60x18 amount = pricing.multiplier.mul(convert(pricing.deliveryCost + costs.foreign));
```

Comment:
Since we do not have separate multipliers for `amount` and `exchangeRate` and apply the same multiplier to both, would it be better to only apply the multiplier once.

 In `_convertToNative` we essentially do `amount * exchangeRate`. 
So when we apply the multiplier `m` yields the substitution:
`(m * amount) * (m * exchangeRate)`
Or simplified:
`m^2 * amount * exchangeRate`

This makes the calculation nonlinear because of the multiplier squared. Setting the multiplier to `3` will have the effect of multiplying by `9` on the final value.

Would it not be simpler to make this linear by just scaling the result of `_convertToNative(exchangeRate, amount)` by the multiplier once. This will make this parameter easy to reason about and harder to configure incorrectly.

This also looks like how it is implemented on the other side of the bridge:
https://github.com/Snowfork/polkadot-sdk/pull/127/files#diff-80bccd165d72d122235ef3d186d56f161fcaa5ce73e2a8215987deefc070a2eaR383-R385

Reply:
IIUC the multiplier will be 1.1 or something, so 1.1(for weight)*1.1(for exchangeRate)=1.21, with the extra 21% we're safe.

Reply:
But do we also need to take the fluctuation of `fee_per_gas` into consideration? If so we need to apply the multiplier to https://github.com/Snowfork/polkadot-sdk/blob/c2936d77a4ee208d06d85f5c4df3c2d919b25d97/bridges/snowbridge/pallets/outbound-queue/src/lib.rs#L373-L378

Reply:
> Alistair: Since we do not have separate multipliers for amount and exchangeRate and apply the same multiplier to both, would it be better to only apply the multiplier once.

To echo what Ron says, both the Substrate WeightToFee (and other AssetHub/BridgeHub costs) and the DOT/ETH exchange rate are subject to fluctuations. So we need to multiply them both by the multiplier.

> Ron: Do we need a separate multiplier for the price params on ethereum side?
> For example, take the fluctuation of fee_per_gas into consideration. Then we also need to apply the multiplier to

The `multiplier`parameter is only meant to be used for exchange rates, and substrate execution costs.

The gas price (FeePerGas) is very volatile, and so instead, as I've updated the docs to represent, `PricingParameters.FeePerGas` is now the *maximum* amount of gas  we're willing to compensate relayers for. So you can say that `PricingParameters.FeePerGas` already has a multiplier applied.